### PR TITLE
Faq sec 2

### PR DESCRIPTION
**Pull request by etpalmer63**
## Summary
This is a first draft of a FAQ section for the rst docs. Please edit as you see fit. 

## Checklist

The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [x] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Docs/sphinx_documentation/source/Faq.rst [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+**A.** Use ``The_Pinned_Arena()`` and ``htod_memcpy()`` or ``dtoh_memcpy()``. See
+`Memory Allocation`_ in the AMReX Source Documentation.
+
+.. _`Memory Allocation`: https://amrex-codes.github.io/amrex/docs_html/GPU.html#memory-allocation
+
+|
+
+**Q.** How can you prevent a section of code from running on the GPU for profiling purposes?
+
+**A.**
```
Comment (wyphan) [inline Docs/sphinx_documentation/source/Faq.rst]: Use <RunOn::Host>. See https://amrex-codes.github.io/amrex/docs_html/Basics.html?basefab-farraybox-iarraybox-and-array4 for details

File: Docs/sphinx_documentation/source/Faq.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+
+**Q.** Why am I getting a segmentation fault after my code runs?
+
+**A.** Do you have :cpp:`amrex::initialize(); {` and :cpp:`} amrex::finalize();`
+at the beginning and end of your code? For all AMReX commands to function
+properly, including to release resources, they need to be contained
+between these two curly braces. In the `Initialize and Finalize`_ section,
+these commands are discussed further detail.
+
```
Comment (wyphan) [inline Docs/sphinx_documentation/source/Faq.rst:15]: Also, don't forget to pass the input file as the first argument.

File: Docs/sphinx_documentation/source/Faq.rst [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+**Q.** How do I know I am getting the right answer?
+
+**A.** AMReX provides support for verifying output with several tools. To briefly mention a few:
+
+- The :cpp:`print_state` command can be used to output the data of a single cell.
+- :cpp:`VisMF::Write` can be used to write MultiFab data to disk that can be viewed with `Amrvis`_.
+- :cpp:`amrex::Print()` is useful for printing
+  output when using multiple processes or threads as it prevents messages
+  from getting mixed up.
+
```
Comment (wyphan) [inline Docs/sphinx_documentation/source/Faq.rst:65]: Mention `fcompare` utility too, perhaps?

Comment (etpalmer63) [reply Docs/sphinx_documentation/source/Faq.rst:65]: Yes and I'll link to its new entry in the Post-Processing section.

File: Docs/sphinx_documentation/source/Faq.rst [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+device to host copies without relying on managed memory?
+
+**A.** Use ``The_Pinned_Arena()`` and ``htod_memcpy()`` or ``dtoh_memcpy()``. See
+`Memory Allocation`_ in the AMReX Source Documentation.
+
+.. _`Memory Allocation`: https://amrex-codes.github.io/amrex/docs_html/GPU.html#memory-allocation
+
+|
+
+**Q.** How can I prevent a section of code from running on the GPU?
```
Comment (WeiqunZhang) [inline Docs/sphinx_documentation/source/Faq.rst:128]: I don't think this question is suitable for FAQ.

Comment (etpalmer63) [reply Docs/sphinx_documentation/source/Faq.rst:128]: Can we change the question to something that would be more suitable but still close? Or can we stress something in the answer, like "This should only be used for debugging purposes"  that would make it better?

File: Docs/sphinx_documentation/source/Faq.rst (hunk 2 above)
Comment (etpalmer63) [reply Docs/sphinx_documentation/source/Faq.rst:15]: I think not including the input file gives a seg fault immediately. But please let me know if I'm mistaken.

Comment (wyphan) [reply Docs/sphinx_documentation/source/Faq.rst:15]: For some codes, yes. For some others, it gives an abort instead. In any case, the error message emitted might not be obvious to the user.

**Pull request closed without merging**